pkg/cmd/cli/restic/server.go

```diff
@@ -191,3 +202,50 @@ func (s *resticServer) run() {
 	s.logger.Info("Waiting for all controllers to shut down gracefully")
 	wg.Wait()
 }
+
+// validatePodVolumesHostPath validates that the pod volumes path contains a
+// directory for each Pod running on this node
+func (s *resticServer) validatePodVolumesHostPath() error {
+	files, err := s.fileSystem.ReadDir("/host_pods/")
+	if err != nil {
+		return errors.Wrap(err, "could not read pod volumes host path")
+	}
+
+	// create a map of directory names inside the pod volumes path
+	dirs := sets.NewString()
+	for _, f := range files {
+		if f.IsDir() {
+			dirs.Insert(f.Name())
+		}
+	}
+
+	pods, err := s.kubeClient.CoreV1().Pods("").List(metav1.ListOptions{FieldSelector: fmt.Sprintf("spec.nodeName=%s", os.Getenv("NODE_NAME"))})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	valid := true
+	for _, pod := range pods.Items {
```

Comment:
I'm thinking that we might want to ignore pods that don't currently have a phase of `Running`, since there could be pods that have been assigned a node name by the scheduler but haven't yet been initialized on the node, that would cause this check to fail.